src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -308,19 +308,42 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
                 return null;
             }
 
-            SNIHandle sniHandle = null;
+            if (isIntegratedSecurity)
+            {
+                string hostName = details.ServerName;
+                int connPort = details.Port;
+                string connInstanceName = details.InstanceName;
+
+                try
+                {
+                    if (connPort >= 0)
+                    {
+                        spnBuffer = GetSqlServerSPN(hostName, connPort);
+                    }
+                    else
+                    {
+                        spnBuffer = GetSqlServerSPN(hostName, connInstanceName);
+                    }
+                }
+                catch (Exception e)
+                {
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.TCP_PROV, SNICommon.InvalidConnStringError, e);
```

Comment:
The error should be for INVALID_PROV and not TCP_PROV. SPN fetching is provider agnostic.

This shouldn't be a InvalidConnectionStringError. It would be better to not have any custom error text and simply propagate the `e` to the SNIError which will have the reason about what failed. 

`SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.INVALID_PROV, string.empty, e);`